### PR TITLE
[Rust Frontend][gRPC] Add audio and video media inputs

### PR DESCRIPTION
**Pull request by connorcarpenter15**
## Purpose

- Accept audio and video HTTP(S) URLs, data URIs, and raw bytes in Rust frontend gRPC inference while preserving UUID and MIME metadata.
- Preserve video structural-prefix ranges and Qwen3-ASR audio positions through prefill/decode.
- No duplicate open PR was found; adjacent work covers preprocessed features or media-domain policy, not this conversion path.

## Test Plan

- `cargo fmt --all -- --check`
- `cargo test -p vllm-server` and `cargo test -p vllm-chat`
- `ruff format --check`, `ruff check`, and `pytest --confcutdir=tests/model_executor tests/model_executor/test_qwen3_asr_mrope.py -q`
- GPU aggregate and prefill/decode validation with `Qwen/Qwen3-VL-8B-Instruct` and `Qwen/Qwen3-ASR-1.7B`.

## Test Result

- Formatting and lint passed; 344 server tests, 311 chat tests, and the focused Python test passed.
- On two A100 80 GB GPUs:
  - Video, using the documented sample MP4: aggregate and prefill/decode returned HTTP 200, a non-empty completion, terminal `length`, and 2,292 prompt tokens from a 12-token text prompt.
  - Audio, using the official `winning_call.ogg` sample and model-native audio placeholder tokens: aggregate and prefill/decode returned HTTP 200, a non-empty transcription, terminal `length`, and 346 prompt tokens from a 13-token text prompt.
- All four GPU runs exited successfully with no gRPC `UNIMPLEMENTED` or media-preparation errors.

**AI assistance disclosure:** This PR was authored with AI assistance and reviewed by the submitter.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
@codex review

File: vllm/model_executor/models/qwen3_asr.py [hunk 1]
```diff
@@ -574,13 +574,10 @@ def get_mrope_input_positions(
...
-            audio_feature_length = mm_feature.data["audio_feature_lengths"].data
-            if isinstance(audio_feature_length, torch.Tensor):
-                audio_feature_length = audio_feature_length.item()
-            audio_len = _get_feat_extract_output_lengths(
-                torch.tensor(audio_feature_length)
-            ).item()
+            # The placeholder already contains one token per audio embedding.
+            # Use its length so M-RoPE can be reconstructed even when a covered
+            # prefix no longer carries the audio processor tensors.
+            audio_len = mm_feature.mm_position.length
```
Comment (Isotr0py) [inline vllm/model_executor/models/qwen3_asr.py:581]: Can you check the ASR wef score with https://huggingface.co/datasets/openslr/librispeech_asr? I want to double check the accuracy to avoid potiential edge case. Also the multi-audio cases should be validated.

You can easily run the benchmark with `evalscope`:
https://github.com/modelscope/evalscope/blob/54813f5c00d7b3dd02ccbdc78d382c019036b178/docs/en/benchmarks/librispeech.md#L4

Comment (connorcarpenter15) [reply vllm/model_executor/models/qwen3_asr.py:581]: Hi @Isotr0py, I ran `Qwen/Qwen3-ASR-1.7B` over the complete `openslr/librispeech_asr`:
- Aggregate WER: 1.943%; 2,620/2,620 requests succeeded.
- Prefill/decode WER: 1.958%; 2,620/2,620 requests succeeded.

I also tested one request containing two unequal-length audio inputs. Both were transcribed completely and in input order in aggregate and prefill/decode.

**Review by Isotr0py — APPROVED** [review]
